Editor/AGS.Types/Interfaces/IRoomController.cs

```diff
@@ -21,26 +21,76 @@ public interface IRoomController
+        /// </summary>
+        /// <param name="background">The background index to delete.</param>
+        void DeleteBackground(int background);
+        /// <summary>
+        /// Gets the loaded room specified mask as a <see cref="Bitmap"/> instance.
+        /// </summary>
+        /// <param name="mask">The mask type to get.</param>
+        /// <returns>A <see cref="Bitmap"/> instance with the selected mask type.Return null if none is selected.</returns>
+        Bitmap GetMask(RoomAreaMaskType mask);
+        /// <summary>
+        /// Sets the loaded room specified mask. This change is not permanent until <see cref="IRoomController.Save"/>
+        /// is invoked.
+        /// </summary>
+        /// <param name="mask">The mask type to set.</param>
+        /// <param name="bmp">The mask to set.</param>
+        void SetMask(RoomAreaMaskType mask, Bitmap bmp);
+        /// <summary>
+        /// Gets the area number on the specified room mask at (x,y)
+        /// RequiredAGSVersion: 3.0.1.35
+        /// </summary>
+        int  GetAreaMaskPixel(RoomAreaMaskType maskType, int x, int y);
+        /// <summary>
+        /// Draws the room background to the specified graphics context.
+        /// RequiredAGSVersion: 3.0.1.35
+        /// </summary>
+        [Obsolete("The method is deprecated because it takes integer for scale which is inaccurate. Use overload with double for scale instead.")]
+        void DrawRoomBackground(Graphics g, int x, int y, int backgroundNumber, int scaleFactor);
 		/// <summary>
 		/// Draws the room background to the specified graphics context,
 		/// and overlays one of the room masks onto it.
 		/// RequiredAGSVersion: 3.0.1.35
 		/// </summary>
+        [Obsolete("The method is deprecated because it takes integer for scale which is inaccurate. Use overload with double for scale instead.")]
 		void DrawRoomBackground(Graphics g, int x, int y, int backgroundNumber, int scaleFactor, RoomAreaMaskType maskType, int maskTransparency, int selectedArea);
-		/// <summary>
-		/// Sets whether or not to grey out non-selected masks when drawing the background.
-		/// </summary>
-		bool GreyOutNonSelectedMasks { set; }
+        /// <summary>
+        /// Draws the room background to the specified graphics context,
+        /// and overlays one of the room masks onto it.
+        /// </summary>
+        void DrawRoomBackground(Graphics g, Point point, int backgroundNumber, double scaleFactor, RoomAreaMaskType maskType, int maskTransparency, int selectedArea);
+        /// <summary>
+        /// Scales all the room's masks according to the <see cref="Room.MaskResolution"/>,
+        /// execept for the <see cref="RoomAreaMaskType.WalkBehinds"/> which retains the
+        /// resolution of the background image. If the <see cref="Room.MaskResolution"/>
+        /// has a resolution of 2 it will halve the masks in both dimensions.
+        /// </summary>
+        void AdjustMaskResolution();

```

Comment:
Maybe this shouldn't be a part of the public interface? Maybe do an automatic scale when room mask resolution is changed instead

Reply:
> Maybe do an automatic scale when room mask resolution is changed instead

Could you elaborate on this?

Reply:
One thing I will agree with is that the way function is made right now seems too obscure. It had to be called manually after you change Room.MaskResolution property, which is not good. Maybe make it accept new resolution factor and then call it and let it set that factor to the room inside the implementation instead?

Reply:
When the mask resolution for a room is changed in the room editor it will invoke the `AdjustMaskResolution()` and scale down or up the actual bitmaps, except for walkbehind mask. The bitmaps will physically change size so if your mask resolution is 1, and then you set mask resolution = 2 for a room with 320x240 dimensions the masks will become 160x120 on disk.

I'm thinking that maybe this isn't the kind of feature we want to expose to a plugin developer?

Reply:
> Maybe make it accept new resolution factor and then call it and let it set that factor to the room inside the implementation instead?

Good idea actually. I think this might be the way to go

Reply:
Changed it